tools/hsmtool.c

```diff
@@ -368,6 +372,148 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
 	return 1;
 }
 
+static void get_words(struct words **words) {
+	printf("Select your language:\n");
```

Comment:
There seem to be quite a number of repetitions in the code here, that might be better suited for a list of structs that contain the various pieces of information:

```c
struct wordlist_lang {
	char *abbr;
	char *name;
};

struct wordlist_lang languages[] = {
	{"en", "English"},
	{"es", "Spanish"},
	{"fr", "French"},
	{"it", "Italian"},
	{"jp", "Japanese"},
	{"zhs", "Chinese Simplified"},
	{"zht", "Chinese Traditional"},
};
```
Then you can use `ARRAY_SIZE` to determine the size of the array at compile time, iterate through them when printing the menu, and indexing with the result after selection to find the abbreviation to use when fetching the wordlist from `libwally` instead of having to update the `printf`, index checks and switch-statement below.

Reply:
Updated